### PR TITLE
Form editor: Input styles settings [MAILPOET-2599]

### PR DESCRIPTION
**Pull request by costasovo**
[MAILPOET-2599]

[MAILPOET-2599]: https://mailpoet.atlassian.net/browse/MAILPOET-2599

### CODE REVIEW

File: assets/js/src/form_editor/blocks/input_styles_settings.jsx [hunk 1]
```diff
@@ -1,16 +1,61 @@
-import React from 'react';
+import React, { useState } from 'react';
 import MailPoet from 'mailpoet';
 import {
   Panel,
   PanelBody,
+  ToggleControl,
 } from '@wordpress/components';
+import { partial } from 'underscore';
```
Comment (pavel-mailpoet) [inline assets/js/src/form_editor/blocks/input_styles_settings.jsx]: why underscore? I thought we are going to slowly move to `lodash` and remove underscore eventually.

File: assets/js/src/form_editor/blocks/input_styles_settings.jsx [hunk 2]
```diff
@@ -1,16 +1,61 @@
...
+          checked={localStyles.fullWidth}
+          onChange={partial(updateStyles, 'fullWidth')}
+        />
+        <ToggleControl
+          label={MailPoet.I18n.t('formSettingsInheritStyleFromTheme')}
+          checked={localStyles.inheritFromTheme}
+          onChange={partial(updateStyles, 'inheritFromTheme')}
+        />
+        {!localStyles.inheritFromTheme ? (
+          <>
```
Comment (pavel-mailpoet) [inline assets/js/src/form_editor/blocks/input_styles_settings.jsx]: Why do we need the fragment here?

File: assets/js/src/form_editor/store/blocks_to_form_body.jsx [hunk 3]
```diff
@@ -1,5 +1,16 @@
 import { has } from 'lodash';
 
+const mapBlockStyles = (styles) => {
+  const mappedStyles = {
+    full_width: styles.fullWidth ? '1' : '0',
```
Comment (pavel-mailpoet) [inline assets/js/src/form_editor/store/blocks_to_form_body.jsx]: I think we use '' in the database and not '0'

File: assets/js/src/form_editor/blocks/input_styles_settings.jsx (hunk 2 above)
Comment (costasovo) [reply assets/js/src/form_editor/blocks/input_styles_settings.jsx]: There are more elements in this jsx block so we need to wrap them.

File: assets/js/src/form_editor/store/blocks_to_form_body.jsx (hunk 3 above)
Comment (costasovo) [reply assets/js/src/form_editor/store/blocks_to_form_body.jsx]: It seems that we use '0' and we check for '1'. I'll keep '0' although I don't like it. It is consistent with other properties.

**Comment by costasovo** [comment]
@pavel-mailpoet I added 3 new commits

**Comment by costasovo** [comment]
@pavel-mailpoet I've added 5 commit addressing issues mentioned on slack.

**Comment by costasovo** [comment]
@pavel-mailpoet I rebased this PR and fix conflicts + refactor the last commit with the fix for colour pickers

**Comment by veljkho** [comment]
Hi @costasovo, I am unable to merge this PL because of some conflict with a branch or something.. Can you please assist to resolve this? Thanks.

**Comment by costasovo** [comment]
@veljkho Sure. It seems to be in conflict with previously merged headings block PR.

**Comment by costasovo** [comment]
@veljkho I fixed the conflict and rebased the PR.

**Pull request merged**